### PR TITLE
libsonata,py-libsonata: update to v0.1.26

### PR DESCRIPTION
**Pull request by mgeplf**

### CODE REVIEW

**Comment by mgeplf** [comment]
Not sure I did this right; do I edit both of these files when something has been upstreamed?

File: var/spack/repos/builtin/packages/py-libsonata/package.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ class PyLibsonata(PythonPackage):
     maintainers("tristan0x")
 
     version("master", branch="master")
+    version("0.1.25", sha256="b653cbefbc511fe24ccf5cce7d80253954ec280800af5fd33700f0faea93fd4c")
```
Comment (jorblancoa) [inline var/spack/repos/builtin/packages/py-libsonata/package.py]:
```suggestion
    version("0.1.26", sha256="b653cbefbc511fe24ccf5cce7d80253954ec280800af5fd33700f0faea93fd4c")
```

**Comment by mgeplf** [comment]
Looks like we need an update of bluecellulab; I will make a PR.

**Comment by mgeplf** [comment]
@anilbey can you look at the updated py-bluecellulab to make sure the versions look ok?

**Review by matz-e — DISMISSED** [review]

**Comment by anilbey** [comment]
> @anilbey can you look at the updated py-bluecellulab to make sure the versions look ok?

All good, thanks @mgeplf

File: bluebrain/repo-bluebrain/packages/py-bluepysnap/package.py [hunk 2]
```diff
@@ -17,17 +17,21 @@ class PyBluepysnap(PythonPackage):
...
     depends_on("py-cached-property@1.0:", type=("build", "run"))
     depends_on("py-h5py@3.0.1:3", type=("build", "run"))
-    depends_on("py-importlib-resources@5:", when="@2", type=("build", "run"))
+    depends_on("py-importlib-resources@5:", when="@2:", type=("build", "run"))
     depends_on("py-jsonschema@4", type=("build", "run"))
+
     depends_on("py-libsonata@0.1.21:", type=("build", "run"))
-    depends_on("py-libsonata@0.1.24:", when="@2", type=("build", "run"))
+    depends_on("py-libsonata@0.1.24:", when="@2:2", type=("build", "run"))
+    depends_on("py-libsonata@0.1.26:", when="@3:3", type=("build", "run"))
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-bluepysnap/package.py:33]:
```suggestion
    depends_on("py-libsonata@0.1.24:", when="@2", type=("build", "run"))
    depends_on("py-libsonata@0.1.26:", when="@3", type=("build", "run"))
```

**Comment by mgeplf** [comment]
Looks like `bluecellulab` requires pydantic>2:
`bluecellulab 2.6.6 has requirement pydantic<3.0.0,>=2.5.2, but you have pydantic 1.10.9.`
Looking at upstream, they don't have it yet.  IIRC, pydantic requires rust and I don't have the time to update all that at the moment.

File: var/spack/repos/builtin/packages/py-annotated-types/package.py
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-annotated-types/package.py:1]: From upstream?

Note to self: create ticket about upstreaming

Comment (mgeplf) [reply var/spack/repos/builtin/packages/py-annotated-types/package.py:1]: No, I did that one myself.

**Review by matz-e — APPROVED** [review]

**Pull request merged**